### PR TITLE
Allow installation with Symfony v7 components, Remove support for Symfony v5 components

### LINKED ISSUE #117: [RFC]: Allow installation with Symfony 7.x series
## RFC

Allowing installation along with Symfony v7 components will require dropping support for PHP 8.1 here.

Considering that `symfony/console` is an important dependency here, it probably makes sense to do that.

What do members of the @laminas/technical-steering-committee think?

### PR DESCRIPTION
**Pull request by gsteel**
- Closes #117
- Closes #115
- Closes #116

### CODE REVIEW

File: src/Input/AbstractParamAwareInput.php [hunk 1]
```diff
@@ -306,4 +306,9 @@ static function (mixed $value) use ($originalValidator) {
 
         return $originalValidator;
     }
+
+    public function __toString(): string
+    {
+        return '';
+    }
```
Comment (gsteel) [inline src/Input/AbstractParamAwareInput.php]: Currently got no idea what `__toString` here should actually return, or what its purpose is. Please feel free to jump in and add to this patch directly 👍

**Comment by gsteel** [comment]
Looks like we can support both v6 and v7 without breaking BC but v5 needs to be dropped.

File: src/Input/AbstractParamAwareInput.php (hunk 1 above)
Comment (Xerkus) [reply src/Input/AbstractParamAwareInput.php]: It returns parameters passed in a form as it should appear in terminal, I believe. Essentially a reverse to input parsing.
Like `--foo "value"` or `--foo "value" --foo "another-value"` or whatever else input format it uses.

It should not be present here at all imo as it needs to be implemented by final class. As it is now it makes implementation optional with arguably invalid fallback.

File: phpunit.xml.dist [hunk 2]
```diff
@@ -3,19 +3,26 @@
...
+        <testsuite name="laminas-cli">
+            <directory>./test/</directory>
+        </testsuite>
+    </testsuites>
+    <source>
+        <include>
+            <directory suffix=".php">./src</directory>
+        </include>
+        <exclude>
+            <file>./src/autoload.php</file>
```
Comment (Xerkus) [inline phpunit.xml.dist]: This reminds me. Look in the previous releases on how compat between different console versions was achieved.

It is gone now and should be dropped here.

File: src/Input/AbstractParamAwareInput.php (hunk 1 above)
Comment (froschdesign) [reply src/Input/AbstractParamAwareInput.php]: Here are two examples from the console component itself:

- https://github.com/symfony/console/blob/64e06788686633deb8d9a7c75ab31bcf4b233a26/Input/ArrayInput.php#L92-L114
- https://github.com/symfony/console/blob/64e06788686633deb8d9a7c75ab31bcf4b233a26/Input/ArgvInput.php#L345-L363

No idea if it really helps.

**Review by snapshotpl — APPROVED** [review]

**Pull request merged**